### ISSUE TITLE
Build learned query expansion distance model (calibrate synonym expansion)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#711]
## Problem

**Conservative is right, but how conservative?**

Current approach: All-or-nothing synonym expansion
- Either expand to ALL synonyms or don't expand
- Cascading problem: synonyms of synonyms retrieve unrelated documents

**Example cascade:**
```
Query: "fondis" 
→ Expand: fond → kre, establ, iniciat
→ CASCADE: establ → firme, solide
→ Result: Retrieve unrelated "firmly establishing" sentences
```

This is why documents ranked 6-20 contain noise.

## Solution: Learned Expansion Distance

**Model task**: Given (query_root, candidate_synonym, query_context) → should we expand?

**Training data**: Use evaluation results
- Positive: Synonym expansions that led to correct retrieval
- Negative: Synonym expansions that added noise (retrieved wrong docs)

**Architecture**: Simple MLP or XGBoost
- Input: Root embeddings for query term + candidate synonym + cosine similarity + query type
- Output: Binary (expand or not)
- Size: ~100K params

**Example learned behavior:**
```
fond → kre (confidence: 0.95) ✓ Expand
fond → establ (confidence: 0.70) ✓ Expand  
establ → firme (confidence: 0.30) ✗ Stop cascade
```

## Why This Aligns with Project Thesis

This is NOT:
- ❌ Replacing deterministic extraction with neural networks
- ❌ Learning grammar rules
- ❌ End-to-end neural QA

This IS:
- ✅ Learning ONE calibration decision: expansion distance
- ✅ Using learned model for UNCERTAINTY (how far is too far?)
- ✅ Keeping structure and logic deterministic

**Use learned capacity for decisions that require data, not rules.**

## Expected Impact

Fix retrieval ranking problem (30% of failures due to answer not retrieved)

## Effort

2-3 days:
- Day 1: Build training data from evaluation results
- Day 2: Train model (MLP or XGBoost)
- Day 3: Integrate into query expansion, evaluate

## Dependencies

- Issue #709 (BM25 analysis) should inform this
- Disable reranker first (#708) to get clean baseline

## Related

Part of QA improvement epic. First learned component to add after maximizing deterministic baseline.

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
